core/src/main/scala/org/apache/spark/api/java/JavaUtils.scala

```diff
@@ -43,6 +43,13 @@ private[spark] object JavaUtils {
 
     override def size: Int = underlying.size
 
+    // Delegate to implementation because AbstractMap implementation iterates over whole key set
+    override def containsKey(key: AnyRef): Boolean = try {
```

Comment:
This reminds me that this whole class could go away once only Scala 2.12 is supported, but that's a while away.

Simpler as this?

```
override def containsKey(key: AnyRef): Boolean = key match {
   case a: A => underlying.contains(a)
   case _ => false
}
```

if so then get() could be:

```
override def get(key: AnyRef): B = key match {
  case a: A => underlying.getOrElse(a, null.asInstanceOf[B])
  case _ => null.asInstanceOf[B]
}
```

What about other methods -- anything worth plumbing directly to the underlying implementation?

Reply:
@srowen 
Yeah, you're right -) I have been focused on code around.  Scala isn't my language haven't known that construction, thx. 

> What about other methods -- anything worth plumbing directly to the underlying implementation?

Yep, checked, remove(key) still plumbing, actually, I don't know how to simply fix it, cuz there are no methods to remove it without recreating a new Map.